lib/syskit/cli/log_runtime_archive_main.rb

```diff
@@ -114,6 +114,51 @@ def transfer_server( # rubocop:disable Metrics/ParameterLists
                               implicit_ftps)
             end
 
+            desc "watch_ensure_free_space", "watches the ensure free space process"
+            option :period,
+                   type: :numeric, default: 10, desc: "polling period in seconds"
+            option :max_size,
+                   type: :numeric, default: 10_000, desc: "max log size in MB"
+            option :free_space_low_limit,
+                   type: :numeric, default: 5_000, desc: "start deleting files if \
+                    available space is below this threshold (threshold in MB)"
+            option :free_space_freed_limit,
+                   type: :numeric, default: 25_000, desc: "stop deleting files if \
+                    available space is above this threshold (threshold in MB)"
+            def watch_ensure_free_space(source_dir)
+                loop do
+                    ensure_free_space(source_dir)
+
+                    puts "Ensured free space in #{source_dir}, " \
+                         "sleeping #{options[:period]}s"
+                    sleep options[:period]
+                end
+            end
+
+            desc "ensure_free_space", "ensures there is free space, if not, start \
+                                       deleting files"
+            option :max_size,
+                   type: :numeric, default: 10_000, desc: "max log size in MB"
+            option :free_space_low_limit,
+                   type: :numeric, default: 5_000, desc: "start deleting files if \
+                    available space is below this threshold (threshold in MB)"
+            option :free_space_freed_limit,
+                   type: :numeric, default: 25_000, desc: "stop deleting files if \
+                    available space is above this threshold (threshold in MB)"
+            def ensure_free_space(source_dir)
+                source_dir = validate_directory_exists(source_dir)
+
+                archiver = make_archiver(source_dir)
+
+                source_dir.children.select(&:directory?).sort_by(&:mtime).each do |child|
+                    archiver.ensure_free_space(
```

Comment:
If you look at a big disk, the command would end up removing very old logs in a folder with a *lot* of folders. It would be a lot better to stop as soon as there's enough free space.

I would recommend to have ensure_free_space return a true/false to indicate whether or not there is enough free space still, and stop if there is.

Reply:
Another thing to keep things clean: this loop or the one for data transfer should delete empty folders.

I'd say the data transfer one, actually. The archive loop should as well. Maybe subject for the next PR.